### PR TITLE
Split CSE/Packer scripts and add Mariner

### PR DESCRIPTION
**Pull request by hbeberman**
This pull request splits cse_helpers.sh, cse_install.sh, and tool_installs.sh into subscripts. Each of those scripts still contain the common functions, and anything that is distro specific has moved to a *_ubuntu.sh and *_mariner.sh subscript.

When deploying the scripts, CSE and Packer will copy the common scripts as normal, and will copy the correct distribution's scripts, and rename them to *_distro.sh.

Wherever these scripts were sourced to set their functions in the environment, they now also source the *_distro.sh versions.

### CODE REVIEW

**Comment by hbeberman** [comment]
This is a continuation of Pull https://github.com/Azure/AgentBaker/pull/594, but from a local branch instead of a fork so that the pipeline tests can run automatically.

**Comment by eriksywu** [comment]
can you add a few unit tests in pkg/agent/baker_test.go for Mariner pools?

File: parts/linux/cloud-init/artifacts/mariner/cse_install_mariner.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    echo "SGX drivers not yet supported for Mariner"
+    exit $ERR_SGX_DRIVERS_START_FAIL
+}
+
+# CSE+VHD can dictate the containerd version, users don't care as long as it works
+installStandaloneContainerd() {
+    # azure-built runtimes have a "+azure" suffix in their version strings (i.e 1.4.1+azure). remove that here.
+    CURRENT_VERSION=$(containerd -version | cut -d " " -f 3 | sed 's|v||' | cut -d "+" -f 1)
+    # v1.4.1 is our lowest supported version of containerd
+    local CONTAINERD_VERSION="1.4.3"
```
Comment (eriksywu) [inline parts/linux/cloud-init/artifacts/mariner/cse_install_mariner.sh:34]: is this the rpm repo we'll be using? https://packages.microsoft.com/cbl-mariner/1.0/prod/base/x86_64/rpms/

I only see moby-containerd-1.3.7.

Comment (hbeberman) [reply parts/linux/cloud-init/artifacts/mariner/cse_install_mariner.sh:34]: Yes that's the repo. I've got a task to bring in a newer version of containerd in the next iteration. For the time being the dnf install below this isn't enforcing the specific version request but I'll re-add it once the newer version is available.

File: parts/linux/cloud-init/artifacts/cis.sh [hunk 2]
```diff
@@ -46,7 +46,11 @@ assignFilePermissions() {
     chmod 600 /etc/shadow- || exit $ERR_CIS_ASSIGN_FILE_PERMISSION
     chmod 600 /etc/group- || exit $ERR_CIS_ASSIGN_FILE_PERMISSION
     chmod 644 /etc/default/grub || exit $ERR_CIS_ASSIGN_FILE_PERMISSION
-    for filepath in /etc/crontab /etc/cron.hourly /etc/cron.daily /etc/cron.weekly /etc/cron.monthly /etc/cron.d; do
+
+    if [[ -f /etc/crontab ]]; then
+        chmod 0600 /etc/crontab || exit $ERR_CIS_ASSIGN_FILE_PERMISSION
+    fi
+    for filepath in /etc/cron.hourly /etc/cron.daily /etc/cron.weekly /etc/cron.monthly /etc/cron.d; do
```
Comment (xuto2) [inline parts/linux/cloud-init/artifacts/cis.sh:53]: notes for myself - need to verify if this cis.sh file is needed or not.

File: parts/linux/cloud-init/artifacts/mariner/cse_install_mariner.sh [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+echo "Sourcing cse_install_distro.sh for Mariner"
+
+removeContainerd() {
+    retrycmd_if_failure 10 5 60 dnf remove -y moby-containerd
+}
+
+installDeps() {
+    dnf_makecache || exit $ERR_APT_UPDATE_TIMEOUT
+    dnf_update || exit $ERR_APT_DIST_UPGRADE_TIMEOUT
+    for apt_package in ca-certificates cifs-utils cracklib ebtables ethtool fuse git iotop iproute ipset iptables jq pam nfs-utils socat sysstat traceroute util-linux xz zip; do
```
Comment (xuto2) [inline parts/linux/cloud-init/artifacts/mariner/cse_install_mariner.sh:12]: super nit: s/apt_package/package

File: parts/linux/cloud-init/artifacts/setup-custom-search-domains.sh [hunk 4]
```diff
@@ -1,6 +1,8 @@
 #!/bin/bash
 set -x
 source {{GetCSEHelpersScriptFilepath}}
+source {{GetCSEHelpersScriptDistroFilepath}}
```
Comment (xuto2) [inline parts/linux/cloud-init/artifacts/setup-custom-search-domains.sh:4]: FYI https://msazure.visualstudio.com/CloudNativeCompute/_workitems/edit/9413409

File: vhdbuilder/packer/install-dependencies.sh [hunk 5]
```diff
@@ -1,7 +1,15 @@
 #!/bin/bash
+
+OS=$(sort -r /etc/*-release | gawk 'match($0, /^(ID_LIKE=(coreos)|ID=(.*))$/, a) { print toupper(a[2] a[3]); exit }')
+UBUNTU_OS_NAME="UBUNTU"
+MARINER_OS_NAME="MARINER"
+
 source /home/packer/provision_installs.sh
+source /home/packer/provision_installs_distro.sh
```
Comment (xuto2) [inline vhdbuilder/packer/install-dependencies.sh:8]: Not your fault - I noticed there is go template syntax in the install file, not sure if this has negative impact to the vhd build, but I think it's regression from the recent containerd work. We should either move those go template syntax out of the install file (move to cse_config.sh), or move those static methods from cse_install.sh to cse_helpers so we don't need to source cse_install.sh here. 
https://msazure.visualstudio.com/CloudNativeCompute/_workitems/edit/9444055

FYI @eriksywu

**Comment by xuto2** [comment]
> can you add a few unit tests in pkg/agent/baker_test.go for Mariner pools?

+1

File: parts/linux/cloud-init/artifacts/mariner/cse_install_mariner.sh (hunk 1 above)
Comment (xuto2) [reply parts/linux/cloud-init/artifacts/mariner/cse_install_mariner.sh:34]: Create a work item to track the work so we don't forget https://msazure.visualstudio.com/CloudNativeCompute/_workitems/edit/9443959

File: pkg/agent/datamodel/types.go [hunk 6]
```diff
@@ -1038,6 +1040,10 @@ func (p *Properties) GetKubeProxyFeatureGatesWindowsArguments() string {
 	return strings.TrimSuffix(buf.String(), ", ")
 }
 
+func (a *AgentPoolProfile) IsMariner() bool {
```
Comment (xuto2) [inline pkg/agent/datamodel/types.go:1043]: replace with an explicit argument in nodebootstrapconfiguration

**Pull request merged**